### PR TITLE
odb: dbu and micron converters

### PR DESCRIPTION
**Pull request by gudeh**
Conversion functions between micron and DBUs.
Added bidirectional conversion functions for both length and area measurements.

### CODE REVIEW

File: src/odb/include/odb/db.h [hunk 1]
```diff
@@ -1177,6 +1177,26 @@ class dbBlock : public dbObject
   ///
   int getDbUnitsPerMicron();
 
+  ///
+  /// Convert a length from database units (DBUs) to microns.
+  ///
+  double dbuToMicrons(const int64_t dbu);
```
Comment (rovinski) [inline src/odb/include/odb/db.h:1183]: I think dbu is a regular 32-bit int? It's only areas that are 64.

**Review by rovinski — COMMENTED** [review]
I had always assumed there was a reason why there wasn't an odb function for this. If it works, then there are many places in the codebase that can be swapped for this

```
dpl/include/dpl/Opendp.h:  double dbuToMicrons(int64_t dbu) const;
dpl/include/dpl/OptMirror.h:  double dbuToMicrons(int64_t dbu) const;
ppl/src/SimulatedAnnealing.h:  double dbuToMicrons(int64_t dbu);
ppl/include/ppl/IOPlacer.h:  double dbuToMicrons(int64_t dbu);
pdn/src/techlayer.h:  double dbuToMicron(int value) const;
rsz/src/RepairDesign.hh:  double dbuToMicrons(int dist) const;
grt/src/fastroute/include/FastRoute.h:  double dbuToMicrons(int64_t dbu);
grt/include/grt/GlobalRouter.h:  double dbuToMicrons(int64_t dbu);
tap/include/tap/tapcell.h:  double dbuToMicrons(int64_t dbu);
ant/include/ant/AntennaChecker.hh:  double dbuToMicrons(int value);
```

File: src/odb/include/odb/db.h (hunk 1 above)
Comment (gudeh) [reply src/odb/include/odb/db.h:1183]: Correct! Further information: https://openroad.readthedocs.io/en/latest/contrib/DatabaseMath.html

File: src/odb/include/odb/db.h [hunk 2]
```diff
@@ -1177,6 +1177,26 @@
...
+
+  ///
+  /// Convert an area from database units squared (DBU^2) to square microns.
+  ///
+  double dbuAreaToMicrons(const int64_t dbu_area);
+
+  ///
+  /// Convert a length from microns to database units (DBUs).
+  ///
+  int micronsToDbu(const double microns);
```
[the remark this answers is not in the document]
Comment (QuantamHD) [reply src/odb/include/odb/db.h:1193]: Please fix

**Review by eder-matheus — APPROVED** [review]

**Pull request merged**